spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java

```diff
@@ -200,18 +200,35 @@ public boolean rmdir(String directory) throws IOException {
 	public boolean exists(String path) throws IOException {
 		Assert.hasText(path, "'path' must not be empty");
 
-		String currentWorkingPath = this.client.printWorkingDirectory();
-		Assert.state(currentWorkingPath != null,
-				"working directory cannot be determined, therefore exists check can not be completed");
 		boolean exists = false;
-
+		InputStream inputStream = null;
 		try {
-			if (this.client.changeWorkingDirectory(path)) {
-				exists = true;
-			}
+			inputStream = this.client.retrieveFileStream(path);
```

Comment:
My real concern is 3 round trips per exists call when 99% of the use cases will be looking for a file, not a directory.